content/en/docs/tasks/job/intra-job-pod-networking-using-pod-hostnames.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Intra-Job Pod Networking Using Pod Hostnames
+content_type: task
+min-kubernetes-server-version: v1.21
+weight: 30
+---
+
+<!-- overview -->
+
+In this example, we will run a job in [indexed completion mode](https://kubernetes.io/blog/2021/04/19/introducing-indexed-jobs/) configured such that
+the pods created by the job can communicate with each other using pod hostnames rather than pod IPs.
+
+Pods within a Job might need to communicate among themselves. They could query the Kubernetes API
+to learn the IPs of the other Pods, but it's much simpler to rely on Kubernetes' built-in DNS resolution.
+Jobs in indexed completion mode automatically set the pod hostname to be in the format of
+`${jobName}-${completionIndex}`, which can be used to deterministically determine
+pod hostnames and enable pod networking *without* needing to create a client connection to
+the Kubernetes control plane to obtain pod hostnames/IPs via API requests. This can be useful
+for use cases where pod networking is required but we don't want to depend on a network 
+connection with the Kubernetes API server.
```

Comment:
I would remove this sentence... It's not a pattern that anybody uses in the microservices world, so it shouldn't be used in batch either.

Reply:
Done